src/libYARP_OS/include/yarp/os/Filesystem.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2006-2019 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+
+#ifndef YARP_OS_FILESYSTEM_H
+#define YARP_OS_FILESYSTEM_H
+
+namespace yarp {
+namespace os {
+namespace filesystem {
+#if (defined _WIN32 && _MSC_VER >= 1920)
+typedef wchar_t             value_type;
+static constexpr value_type preferred_separator = L'\\';
+static constexpr value_type path_separator = L';';
```

Comment:
What is the rationale of using a different separator just for Visual Studio 2019? Furthermore, it is perfectly fine to use YARP compiled with VS2017 in a  compilation unit compiled with VS2019. Do would mean that for those compilation units `yarp::os::Network::getDirectorySeparator() != yarp::os::filesystem::preferred_separator`, it that ok?

Reply:
The reason is explained here https://github.com/robotology/yarp/pull/2062#issuecomment-513266676 .

In VS 2017 it doesn't compile, it doesn't like the `wchar_t` as argument of constructor of the `std::string`.

Reply:
But I don't understand, this will mean that `/` will be returned for VS2017, while `getDirectorySeparator` returns `\`, if we change  this we need to at least document it in the changelog, no?

Reply:
Ah sorry I see now what you meant!
No there is an error in the if-else statement here, also for vs2017 should be `\`.

Reply:
This should be fixed in the latest version of the patch. The code is now:

```c++
#if (defined _WIN32)
#  if (_MSC_VER >= 1920)
typedef wchar_t             value_type;
static constexpr value_type preferred_separator = L'\\';
static constexpr value_type path_separator = L';';
#  else
typedef char                value_type;
static constexpr value_type preferred_separator = '\\';
static constexpr value_type path_separator = ';';
#  endif
#else
typedef char                value_type;
static constexpr value_type preferred_separator = '/';
static constexpr value_type path_separator = ':';
#endif
```